lib/src/front_end/ast_node_visitor.dart

```diff
@@ -1159,11 +1177,24 @@ class AstNodeVisitor extends ThrowingAstVisitor<Piece> with PieceFactory {
 
   @override
   Piece visitLogicalOrPattern(LogicalOrPattern node) {
-    // If a logical or pattern is the outermost pattern in a switch expression
-    // case, we want to format it like parallel cases and not indent the
-    // subsequent operands.
-    var indent = node.parent is! GuardedPattern ||
-        node.parent!.parent is! SwitchExpressionCase;
+    // If a logical and pattern occurs inside a map pattern entry, we want to
+    // format the operands in parallel, like:
+    //
+    //     var {
+    //       key:
+    //         operand1 &&
+    //         operand2,
+    //     } = value;
+    //
+    // Also, if it's the outermost pattern in a switch expression case, we
+    // flatten the operands like parallel cases:
```

Comment:
Why don't we do the same with `&&`?

Reply:
Good question. We definitely *could.* I added special formatting for `||` because switch expressions don't allow multiple cases to share a body, so you end up using `||` instead and I wanted the resulting formatted code to look like parallel cases. That same logic doesn't really apply to `&&`.

Do you think it would be more readable to format `&&` the same way? For what it's worth, the current formatting style has this same special rule for `||` but not `&&`.

Reply:
I think I might be surprised/confused if I saw a chain of `&&` next to a chain of `||` and they were formatted differently. I haven't noticed that the current formatting differs between them. That's probably because `&&` is rare in that position - and I don't think I've _ever_ seen a chain with more than one `&&` in a pattern in real code.

So maybe it doesn't matter then. I was surprised by the difference as I read it here, but I likely won't ever be surprised by the difference in real code.

Reply:
> I likely won't ever be surprised by the difference in real code.

I think that's true. I'll leave it as it is for now, and if users request the same formatting for `&&`, we can consider it then.